src/main/kotlin/com/yapp/brake/group/dto/request/UpdateGroupAppRequest.kt

```diff
@@ -0,0 +1,15 @@
+package com.yapp.brake.group.dto.request
+
+import jakarta.validation.constraints.NotBlank
+import jakarta.validation.constraints.NotNull
+import jakarta.validation.constraints.PositiveOrZero
+import jakarta.validation.constraints.Size
+
+data class UpdateGroupAppRequest(
+    @field:NotNull
+    @field:PositiveOrZero
+    val groupAppId: Long,
```

Comment:
그룹앱 업데이트할때, 기존에 없던 앱을 추가하는 경우에는 goupAppI을 0으로 받아서 저장하는 방식으로 이해했습니다!

post에서는 groupAppId가 필요없었는데 여기서 groupAppId를 받으니까 뭘 넣어야하는지 헷갈리더라구요 🥲
그래서 문서화에 해당 내용을 추가하거나, 아래와 새로운 앱은 name만 받는 방식은 어떠신가요 ??

```suggestion
    @field:PositiveOrZero
    val groupAppId: Long = 0L,
```

Reply:
> 그룹앱 업데이트할때, 기존에 없던 앱을 추가하는 경우에는 goupAppI을 0으로 받아서 저장하는 방식으로 이해했습니다!

아연님이 이해하신게 맞아요!

> post에서는 groupAppId가 필요없었는데 여기서 groupAppId를 받으니까 뭘 넣어야하는지 헷갈리더라구요 🥲

여기에 대해 부가 설명을 드리면, 디자인을 참고했을 때 수정을 하게 되면 기존 앱, 추가할 앱이 모두 들어있어야 한다고 생각하고 만들었습니다.
왜냐면 삭제된 앱은 여기에 포함되어 있지 않으니 DB에서 그룹 id로 조회 했을 때 삭제된 관리 앱을 알 수 있습니다. 


<img width="256" height="541" alt="image" src="https://github.com/user-attachments/assets/ab0f1f67-f71f-47ef-80bb-e4288dbc3728" />


그런데 아연님 말씀처럼 생성과, 수정 dto를 확인했을때 헷갈린다는 점이 공감이 되네요. 수정을 하게 되면 그럼 삭제할 앱 목록 DTO(Id 포함)와 수정할 앱 목록 DTO(id 미포함)을 나눠서 받는게 좋을까요? 그리고 삭제 되지 않은 앱은 받지는 않고 API 응답에 포함을 할지 말지도 고민이 되네요. 일단 PUT 메서드의 특성상 리소스를 아예 대체한다는 의미를 가져서 전부를 보여주는 쪽으로 생각이 좀 기울 긴 합니다!

> The PUT method requests that the state of the target resource be created or replaced with the state defined by the representation enclosed in the request message payload. -RFC 공식문서-

Reply:
초안은 이렇습니다!

```kotlin
data class UpdateGroupRequest(
    @field:NotBlank
    @field:Size(min = 2, max = 10)
    val name: String,
    @field:Valid
    val addedGroupApps: List<AddGroupAppRequest> = emptyList(),
    val deletedGroupAppIds: List<Long> = emptyList(),
)

```

Reply:
음 일단 현재 구현하신 코드에서 제일 수정이 적은 방안으로 제안드렸던 건데요! 그래서 사실 저는 새로운 앱에 대해 Id가 0이어야 한다는 명세만 명확하게 나오면 현재 상태도 좋다고 생각합니다 😄

그리고 현재 방식이 말씀하셨던 PUT 메서드에서 리소스 전부를 보여주는 방법이라고도 생각이 드네요 ㅎㅎ